src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,153 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int NO_INDEX = -1;
+    private static final double DEFAULT_LOAD_FACTOR = 1.5;
+    private int size;
+    private int currentCapacity = DEFAULT_CAPACITY;
+    private T[] elementData;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        elementData[size] = value;
+        size++;
+        growIfArrayIsFull();
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException(
+                    "Requested index " + index + " is out of bounds");
+        } else {
+            System.arraycopy(elementData, index, elementData, index + 1, size - index);
+            elementData[index] = value;
+            size++;
+            growIfArrayIsFull();
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        Object[] buffer = toArray(list);
+        Object[] elementData;
```

Comment:
This local variable 'elementData' is redundant because you have a class field with the same name. You can directly use 'this.elementData' in the 'System.arraycopy' method on line 45.